### PR TITLE
Bump to .NET 6.0

### PR DESCRIPTION
**Pull request by RussKie**
.NET 5 is nearing EOL, and .NET 6 is LTS.

### CODE REVIEW

**Comment by mstv** [comment]
+1
It seems to work locally including automated tests but again not on AppVeyor. Still `draft`? 
I have reverted AppVeyor back to use VS 2022 and have addressed a few nitpicks.

Since the installation of VS 2022 and .NET6, a new GH sign-in dialog popped up - affects the .NET5 builds, too.

**Comment by RussKie** [comment]
Thank you @mstv!

One thing I struggle to explain or pinpoint the culprit is why UI tests now run much longer than on .NET 5:
[net5.0 build](https://ci.appveyor.com/project/gitextensions/gitextensions/builds/42468965): "_... Duration: **2 m 53 s** - UI.IntegrationTests.dll (net5.0)_"
[net 6.0 build](https://ci.appveyor.com/project/gitextensions/gitextensions/builds/42471732): "..._Duration: **6 m 5 s** - UI.IntegrationTests.dll (net6.0)_"

**Comment by mstv** [comment]
The integration test `RunScript_with_arguments_with_s_option_with_RevisionGrid_without_selection_shall_display_error_and_return_false` hung when trying to open Browse for `C:\`.
The FormBrowse showed the Dashboard instead of "not a repo".
After opening one of the recent repos, the tests continued (but failed with the errors addressed / ignored by #9856).

This test creates an empty temp directory now (with many lines of code).
BTW: The function `Path.GetTempFileName()` of .NET6(?) sucks because it returns names of existing files.

And I have removed an obsolete(?) file from the installer:
```
C:\projects\gitextensions\Setup\Product.wxs(194): error LGHT0103: The system cannot find the file 'C:\projects\gitextensions\artifacts\Release\publish\GitExtensions\\stdole.dll'.
```

**Comment by RussKie** [comment]
[This build](https://ci.appveyor.com/project/gitextensions/gitextensions/builds/42471834) took 12 minutes: "..._Duration: **12 m** - UI.IntegrationTests.dll (net6.0)_"

**Comment by RussKie** [comment]
> And I have removed an obsolete(?) file from the installer:
> 
> ```
> C:\projects\gitextensions\Setup\Product.wxs(194): error LGHT0103: The system cannot find the file 'C:\projects\gitextensions\artifacts\Release\publish\GitExtensions\\stdole.dll'.
> ```

👍

**Comment by RussKie** [comment]
> BTW: The function `Path.GetTempFileName()` of .NET6(?) sucks because it returns names of existing files.

If you have a repro, please raise a bug in https://github.com/dotnet/runtime/.
I think I may have discovered a bug with XmlSerializer which necessitated the change to ResourceManager/Xliff/TranslationSerializer.cs, but I yet to test it in a standalone repo

**Comment by mstv** [comment]
That testcase `RunScript_with_arguments_with_s_option_with_RevisionGrid_without_selection_shall_display_error_and_return_false` passed but took almost 9.5 minutes.

File: ResourceManager/Xliff/TranslationSerializer.cs [hunk 1]
```diff
@@ -9,9 +9,13 @@ public static class TranslationSerializer
     {
         public static void Serialize(TranslationFile translation, string path)
         {
-            using TextWriter tw = new StreamWriter(path, false);
```
Comment (RussKie) [inline ResourceManager/Xliff/TranslationSerializer.cs:12]: A workaround for a regression in .NET 6, see: https://github.com/dotnet/runtime/issues/64885

**Comment by vbjay** [comment]
> > BTW: The function `Path.GetTempFileName()` of .NET6(?) sucks because it returns names of existing files.
> 
> If you have a repro, please raise a bug in https://github.com/dotnet/runtime/.
> I think I may have discovered a bug with XmlSerializer which necessitated the change to ResourceManager/Xliff/TranslationSerializer.cs, but I yet to test it in a standalone repo


Nope. It's done that since day 1

Creates a uniquely named, zero-byte temporary file on disk and returns the full path of that file.

**Comment by mstv** [comment]
FormBrowse showing the Dashboard did not become idle not before I opened a repo via RDesktop:

```
22:22:28.9563908 switching to UI thread
22:22:28.9564016 waiting for idle
22:22:28.9564508 showing form
22:25:57.5378374 idle
FormBrowse available
22:25:57.7819115 form shown
```

It hangs in `UITests.RunForm<T>` at `await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();`.
CC: @sharwell

**Comment by RussKie** [comment]
I think it's good to merge. Any objections?

**Comment by RussKie** [comment]
I don't know why tests are slower, I had more time to investigate

**Comment by gerhardol** [comment]
Pushed an update that fixes build issue for me locally, it was already disabled in AppVeyor so it should not fix that issue (but 6 hanged builds was followed by 2 successes after this change).

Hanging in AppTitle tests
image

**Comment by RussKie** [comment]
> Hanging in AppTitle tests
> image

For me, it appears `RunScript_with_arguments_with_s_option_with_RevisionGrid_without_selection_shall_display_error_and_return_false` is causing the issue:
image

This test expects the revision grid view, however it's "stuck" in the dashboard view. Looking further into it...

**Comment by RussKie** [comment]
I don't believe this test is valid any longer - if the provided path doesn't contain a valid git repo we no longer show the revision grid and show the dashboard.

**Comment by mstv** [comment]
> I don't believe this test is valid any longer - if the provided path doesn't contain a valid git repo we no longer show the revision grid and show the dashboard.

`RunScript_with_arguments_with_s_option_with_RevisionGrid_without_selection_shall_display_error_and_return_false`
has not become invalid - but perhaps the test setup - because the RevisionGrid *can* have _no selection_.
E.g. create a new orphan branch (and do not commit to it yet) - works better with #9919 merged.

**Comment by RussKie** [comment]
> `RunScript_with_arguments_with_s_option_with_RevisionGrid_without_selection_shall_display_error_and_return_false`
> has not become invalid - but perhaps the test setup - because the RevisionGrid _can_ have _no selection_.

Good point, reverted the removal and updated the test arrangement.

**Comment by RussKie** [comment]
Looks like the build is back to ~9-10 min mark

**Comment by RussKie** [comment]
Any further thoughts/objections? If not, I'm going to merge it in a day or two.

**Comment by gerhardol** [comment]
> Any further thoughts/objections? If not, I'm going to merge it in a day or two.

Seem OK to me - we should ship a 4.0 alpha though....

**Pull request merged**